cloudinit/config/modules.py

```diff
@@ -241,8 +237,18 @@ def _run_modules(self, mostly_mods: List[ModuleDetails]):
                 myrep = ReportEventStack(
                     name=run_name, description=desc, parent=self.reporter
                 )
+                func_args = {
+                    "name": name,
+                    "cfg": self.cfg,
+                    "cloud": cc,
+                    "args": args,
+                }
 
                 with myrep:
+                    func_signature = signature(mod.handle)
+                    func_params = func_signature.parameters
+                    if len(func_params) == 5:
+                        func_args.update({"log": LOG})
```

Comment:
I think we must warn users about the fact that we will drop support for modules with the log parameter. Something in the lines of:

```suggestion
                        LOG.warning("DEPRECATED: The use of modules with a `log` parameter is deprecated.")
                        func_args.update({"log": LOG})
```

Reply:
After 6100fda632f63605981636c0900e1e0b8b354979 being merged, I think we might want to use the standard machinery to deprecated things.